sound/soc/sof/intel/hda-dai.c

```diff
@@ -414,6 +414,48 @@ static struct snd_soc_cdai_ops sof_probe_compr_ops = {
 #endif
 #endif
 
+static int ssp_dai_hw_params(struct snd_pcm_substream *substream,
+			     struct snd_pcm_hw_params *params,
+			     struct snd_soc_dai *dai)
+{
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct snd_soc_component *component = snd_soc_rtdcom_lookup(rtd, SOF_AUDIO_PCM_DRV_NAME);
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *sof_dai;
+	struct sof_ipc_reply reply;
+	int ret;
+
+	list_for_each_entry(sof_dai, &sdev->dai_list, list) {
+		if (!sof_dai->cpu_dai_name)
+			continue;
+
+		if (!strcmp(dai->name, sof_dai->cpu_dai_name) &&
+		    substream->stream == sof_dai->comp_dai.direction) {
+			config = &sof_dai->dai_config[sof_dai->current_config];
+			if (!config) {
```

Comment:
This check does not make much sense.
I think you want to check if sof_dai->dai_config is not NULL, I hope that sof_dai->current_config is granted to be sanely between 0 and number_configs?

Reply:
hmm, I'm still puzzled by some legacy processing we in topology.c where some number_config is 0 but there still appears to be a config... anyway your suggestion is good, will change. And anyway I'm setting the current_config so it should be good.